packages/hydrojudge/src/judge/hack.ts

```diff
@@ -54,17 +58,38 @@ export async function judge(ctx: Context) {
         } else if (memory > ctx.config.memory * 1024) {
             status = STATUS.STATUS_MEMORY_LIMIT_EXCEEDED;
         } else {
-            ({ status, message } = await checkers[ctx.config.checker_type]({
-                execute: checker.execute,
-                copyIn: checker.copyIn || {},
-                input: { src: input },
-                output: { content: '' },
-                user_stdout: res.fileIds.stdout ? { fileId: res.fileIds.stdout } : { content: '' },
-                user_stderr: { fileId: res.fileIds.stderr },
-                score: 100,
-                detail: ctx.config.detail ?? true,
-                env: { ...ctx.env, HYDRO_TESTCASE: '0' },
-            }));
+            const stdResult = ctx.config.std ? await runQueued(
+                std.execute,
+                {
+                    stdin: { src: input },
+                    copyIn: std.copyIn,
+                    time: parseTimeMS(ctx.config.time || '1s'),
+                    memory: parseMemoryMB(ctx.config.memory || '256m'),
+                    cacheStdoutAndStderr: true,
+                },
+            ) : { status: STATUS.STATUS_ACCEPTED, fileIds: { stdout: null } };
+            if (stdResult.status === STATUS.STATUS_ACCEPTED) {
+                ({ status, message } = await checkers[ctx.config.checker_type]({
+                    execute: checker.execute,
+                    copyIn: checker.copyIn || {},
+                    input: { src: input },
+                    output: stdResult.fileIds.stdout ? { fileId: stdResult.fileIds.stdout } : { content: '' },
+                    user_stdout: res.fileIds.stdout ? { fileId: res.fileIds.stdout } : { content: '' },
+                    user_stderr: { fileId: res.fileIds.stderr },
+                    score: 100,
+                    detail: ctx.config.detail ?? true,
+                    env: { ...ctx.env, HYDRO_TESTCASE: '0' },
+                }));
+                if (status !== STATUS.STATUS_ACCEPTED && ctx.config.std) {
+                    const tmp = path.join(tmpdir(), `${ctx.request.rid}.out`);
+                    await client.getFile(stdResult.fileIds['stdout'], tmp);
+                    const filename = `hack-${ctx.request.rid.toString()}.out`;
+                    await ctx.session.postFile(ctx.request.rid.toString(), filename, tmp);
```

Comment:
It's not always required to add it to testcases (for example in a contest)
and it will cause hack to fail if postFile was rejected (file count/size limit, etc)